### ISSUE TITLE
Option to only provide "trunk" model?

### ISSUE THREAD

**Issue by zabzug-pfpt** [KevinMusgrave/pytorch-metric-learning#8]
In some cases where a pretrained network (or already-validated architecture) can't be utilized, it's onerous to take a single custom-built network and arbitrary split it into "trunk" and "embedder" pieces.

Would it be possible to modify the Trainer class to allow the user to only provide a single model and a single optimizer?

**Comment by KevinMusgrave** [comment]
Good idea, in the next version I'll make the BaseTrainer check if an "embedder" key exists, and if it does not exist, I'll set the embedder to be Identity(). 

In the meantime, you can do this instead of splitting your custom-built network
```python
class Identity(nn.Module):
    def __init__(self):
        super().__init__()
    def forward(self, x):
        return x


trunk = custom_net()
embedder = Identity()
models = {"trunk": trunk, "embedder": embedder}

# you don't need to supply an embedder optimizer
optimizers = {"trunk_optimizer": trunk_optimizer}
```

**Comment by zabzug-pfpt** [comment]
Ah, that's great! I didn't realize that you could omit the embedder optimizer!

**Comment by KevinMusgrave** [comment]
In the latest version (0.9.74) you can do this for training:
```python
trunk = custom_net()
models = {"trunk": trunk} #leave out embedder
optimizer = {"trunk_optimizer": trunk_optimizer}
```
Similarly, for testing:
```python
tester.test(dataset_dict, epoch, trunk) #leave out embedder
```
